tests/components/squeezebox/conftest.py

```diff
@@ -104,6 +129,141 @@ def mock_setup_entry() -> Generator[AsyncMock]:
         yield mock_setup_entry
 
 
+@pytest.fixture
+def mock_config_entry():
+    """Fixture to create a mock config entry with UUID."""
+
+    def _create():
+        return MockConfigEntry(
+            domain=DOMAIN,
+            unique_id=UUID,
+            data={CONF_HOST: HOST, CONF_PORT: PORT, CONF_HTTPS: False},
+        )
+
+    return _create
+
+
+@pytest.fixture
+def patch_async_query_exception():
+    """Fixture to simulate an exception during Server.async_query."""
+    with patch(
+        "homeassistant.components.squeezebox.config_flow.Server.async_query",
+        side_effect=Exception,
+    ):
+        yield
+
+
+@pytest.fixture
+def mock_async_query():
+    """Fixture to patch Server.async_query with configurable behavior."""
+
+    with patch("pysqueezebox.Server.async_query", autospec=True) as mock:
+
+        def _configure(mode: str):
+            if mode == "uuid":
+                mock.side_effect = None
+                mock.return_value = {"uuid": UUID}
+
+            elif mode == "false":
+
+                async def _cannot_connect(self, *args, **kwargs):
+                    self.http_status = HTTPStatus.BAD_GATEWAY
+                    return False
+
+                mock.side_effect = _cannot_connect
+                mock.return_value = None
+
+            elif mode == "unauthorized":
+
+                async def _unauthorized(self, *args, **kwargs):
+                    self.http_status = HTTPStatus.UNAUTHORIZED
+                    return False
+
+                mock.side_effect = _unauthorized
+                mock.return_value = None
+
+            elif mode == "exception":
+                mock.side_effect = Exception("Unexpected error")
+                mock.return_value = None
+
+            else:
+                raise ValueError(f"Unknown mock_async_query mode: {mode}")
+
+            return mock
+
+        yield _configure
```

Comment:
Okay so 2 things
1. You can update mocks inline. So if we patch out the Server object, we can just do `server_mock.async_query.side_effect = Exception` inline.
2. By doing that, you keep what you expect in the test, which is easier and more versatile. The only reason against doing that would be that if this logic is tested multiple times, you'd have to update every occurence. Which is true, but the downside of making mocks configurable like this is that they get complex real quick. Now there's only 1 parameter, but this could grow to a lot more, which makes it again awkard to extend and harder to modify